code/modules/mob/living/carbon/xenomorph/abilities/shrieker/shrieker_abilities.dm

```diff
@@ -0,0 +1,110 @@
+/datum/action/xeno_action/activable/shriek
+	name = "Shriek"
+	action_icon_state = "screech"
+	ability_name = "Shriek"
+	macro_path = /datum/action/xeno_action/verb/verb_shriek
+	action_type = XENO_ACTION_CLICK
+	ability_primacy = XENO_PRIMARY_ACTION_1
+	xeno_cooldown = 50 SECONDS
+	plasma_cost = 300
+
+	// Config
+
+	// These values are used to determine the
+	// Shriek cost and the switching of different types of shriek
+
+	var/shriek_sound_effectt = "sound/voice/xeno_praetorian_screech.ogg"
+	var/curr_effect_type = SHRIEKER_SHRIEK_BUFF
+
+	var/debuff_daze = 3
+
+	var/shriek_cost = 150
+
+
+/datum/action/xeno_action/onclick/shrieker_switch_shriek_type
+	name = "Toggle Shriek Type"
+	action_icon_state = "roar_motivate" // default = buff
+	macro_path = /datum/action/xeno_action/verb/verb_shrieker_switch_shriek_type
+	action_type = XENO_ACTION_ACTIVATE
+	ability_primacy = XENO_PRIMARY_ACTION_5
+
+/datum/action/xeno_action/onclick/shrieker_switch_shriek_type/can_use_action()
+	var/mob/living/carbon/Xenomorph/X = owner
+	if(X && !X.buckled && !X.is_mob_incapacitated())
+		return TRUE
```

Comment:
instead of overriding parent so that xenos can toggle this ability while resting or stunned, add a variable to xeno_action that when set to TRUE ignores incapacitated checks in /proc/can_use_action